jetty-util/src/main/java/org/eclipse/jetty/util/thread/ReservedThreadExecutor.java

```diff
@@ -212,52 +202,61 @@ public boolean tryExecute(Runnable task)
     {
         if (LOG.isDebugEnabled())
             LOG.debug("{} tryExecute {}", this, task);
-
         if (task == null)
             return false;
 
-        ReservedThread thread = _stack.pollFirst();
-        if (thread == null)
+        boolean offered = _queue.offer(task);
+        int size;
+        while (true)
         {
-            if (task != STOP)
-                startReservedThread();
-            return false;
+            long count = _count.get();
+            size = getLo(count);
+            if (size < 0 || !offered)
+                break;
+            if (_count.compareAndSet(count, getHi(count), --size))
+                break;
         }
-
-        int size = _size.decrementAndGet();
-        if (!thread.offer(task))
-            return false;
-
         if (size == 0 && task != STOP)
```

Comment:
This PR doesn't change the basic logic.  This is still true.

Reply:
OK it changes it a little. Previously if we failed to pick a thread from the stack, it assumed that was because size==0.
This PR now always checks the size, because if it missed the thread because it was late to the rendezvous, then starting another doesn't help and just delays the producing thread more.
So I like this improvement in the logic.  @sbordet

Reply:
I've simplified the loop a little bit.